ca_driver/src/create_driver.cpp

```diff
@@ -222,6 +227,11 @@ void CreateDriver::undockCallback(const std_msgs::EmptyConstPtr& msg)
   robot_->setMode(create::MODE_FULL);
 }
 
+void CreateDriver::mainMotorCallback(const std_msgs::Float32ConstPtr& msg)
+{
+  robot_->setMainMotor(msg->data);
```

Comment:
I like the subscribers over services because of slightly less overhead on the user's end, but perhaps it does make more sense semantically for some functions. I think song/LED requests are good candidates for services, but I'm inclined to leave motor commands as topics (ie. wheel, brush, and vacuum motors) as people may make calls in quick succession to those.